admin/resources/js/components/form/Input.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from "react";
+import { Control, useController } from "react-hook-form";
+
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  /** Object that connects controlled components to react hook form. */
+  control: Control<any>;
+  /** The text for label associated with select input. */
+  label: string;
+  /** A string specifying a name for the input control. */
+  name: string;
+  /** Object set of validation rules to impose on input. */
+  rules?: { [key: string]: any };
+  /** List of validation messages that correspond with the rules list. */
+  ruleMessages?: { [key: string]: string };
+}
+
+const Input: React.FunctionComponent<InputProps> = ({
+  control,
+  label,
+  name,
+  rules = {},
+  ruleMessages,
+  type,
+  ...rest
+}) => {
+  const { field, fieldState } = useController({
+    control,
+    name,
+    rules,
+  });
+
+  return (
+    <div>
+      <label htmlFor={name}>{label}</label>
```

Comment:
Doesn't htmlFor need to match the id of the input, not the name? 

There's no guarantee the name is unique on the page, so we can't always safely use it as id. We could have an optional id field on the Input component, which defaults to using name if not specified? We could append a random string to name to ensure uniqueness? We could nest the input element inside the label element to avoid needing htmlFor, but I think Josh had a good reason to avoid that, unfortunately...

Reply:
Yes, the [for attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#attr-for) needs to match the id value of the input unless the label has a contained controlled element (which it doesn't in this case). I like the random string appending idea: we could use [lodash](https://lodash.com/docs/4.17.15#uniqueId), thoughts on using a util library @tristan-orourke and @yonikid15?

Reply:
I like lodash, its still my favourite general-purpose util library. We could definitely use it to add a random number to the end of an id.
I've also used [short-unique-id](https://www.npmjs.com/package/short-unique-id) before (which as a fun bonus provides some info on the vast unlikelihood of collisions) but its probably overkill.

Reply:
@yonikid15 could you address the uniqueness with lodash then? I know there are no conflicts with naming right now, but it could happen as more elements are added.

Reply:
@mnigh requiring an id prop allows the parent component to address uniqueness in other ways, if necessary.